### PR TITLE
Bug 6974 - Claimant unexpectedly returned to portal

### PR DESCRIPTION
**Pull request by nickwhm**
The bug here relates to using the app on mobile devices. When you change your language settings in the middle of a claim, for example to make sure the date is shown in the correct format in the browser, it refreshes the page. Because it does that it locks the user out of their current claim they are working on. I've added a beforeunload event to save the claim when it refreshes.

### CODE REVIEW

File: src/components/override-sdk/infra/Assignment/Assignment.tsx [hunk 1]
```diff
@@ -310,6 +310,18 @@ export default function Assignment(props) {
...
+    const handleBeforeUnload = () => {
+      buttonPress('saveAssignment', 'secondary');
+    };
+
+    window.addEventListener('beforeunload', handleBeforeUnload);
+    return () => {
+      window.removeEventListener('beforeunload', handleBeforeUnload);
+    };
+  }, []);
+
```
Comment (gautamkumar-hmrc) [inline src/components/override-sdk/infra/Assignment/Assignment.tsx:324]: This is duplicate logic as same is handled in DefaultForm.tsx. One of these need to be removed.

**Comment by nickwhm** [comment]
Closed and re-raisning another PR

File: src/components/override-sdk/infra/Assignment/Assignment.tsx [hunk 2]
```diff
@@ -219,6 +220,8 @@ export default function Assignment(props) {
           const assignmentID = thePConn.getCaseInfo().getAssignmentID();
           const savePromise = saveAssignment(itemKey);
 
+          console.log(`"buttonPress:" ${sAction}, ${sButtonType}`);
```
Comment (AkhilaHMRC) [inline src/components/override-sdk/infra/Assignment/Assignment.tsx:223]: remove the console log

File: src/components/override-sdk/infra/Assignment/Assignment.tsx [hunk 3]
```diff
@@ -303,13 +306,18 @@ export default function Assignment(props) {
   }
   function _onButtonPress(sAction: string, sButtonType: string) {
     buttonPress(sAction, sButtonType);
+    console.log(`"_onButtonPress:" ${sAction}, ${sButtonType}`);
```
Comment (AkhilaHMRC) [inline src/components/override-sdk/infra/Assignment/Assignment.tsx:309]: remove console log

**Review by AkhilaHMRC — CHANGES_REQUESTED** [review]

**Pull request closed without merging**